db/mock/hardware.go

```diff
@@ -0,0 +1,35 @@
+package mock
+
+import (
+	"context"
+)
+
+// DeleteFromDB : delete data from hardware table
+func (d DB) DeleteFromDB(ctx context.Context, id string) error {
```

Comment:
I just have a feedback. When I create a mock I usually use the form:

```go
type DB struct {
    DeleteFromDBFunc func(ctx context.Context, id string) error
    // and all the function like InsertIntoDB should have its InsertIntoDBFunc...
}
```
In this way the real function will use it:

```go
func (d DB) DeleteFromDB(ctx context.Context, id string) error {
	return d.DeleteFromDBFunc(ctx, id)
}
```

And from the test I can inject the behavior I need:

```go
func TestDBReturnsNotFound(t *testing.T) {
    db := DB{
        DeleteFromDBFunc: func(ctx context.Context, id string) error {
            return errros.Error("not found")
        },
    }
    hardware(db)
}
```

Reply:
Addressed in PR #256